config/v1alpha1/types_insights.go

```diff
@@ -48,24 +47,78 @@ type GatherConfig struct {
 	// When set to None the data is not obfuscated.
 	// When set to ObfuscateNetworking the IP addresses and the cluster domain name are obfuscated.
 	// When omitted, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
-	// The current default is None.
 	// +optional
 	DataPolicy DataPolicy `json:"dataPolicy,omitempty"`
 	// disabledGatherers is a list of gatherers to be excluded from the gathering. All the gatherers can be disabled by providing "all" value.
 	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The format for the disabledGatherer should be: {gatherer}/{function} where the function is optional.
+	// Gatherer consists of a lowercase string that may include underscores (_).
+	// Function consists of a lowercase string that may include underscores (_) and is separated from the gatherer by a forward slash (/).
 	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
 	// Run the following command to get the names of last active gatherers:
 	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
 	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	// +kubebuilder:validation:MaxItems=100
+	// +kubebuilder:validation:items:MaxLength=256
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, x.matches("^[a-z]+[_a-z]*[a-z]([/a-z][_a-z]*)?[a-z]$"))`,message=`disabledGatherers must be in the format of {gatherer}/{function} where the gatherer and function are lowercase strings that may include underscores (_) and are separated by a forward slash (/) if the function is provided`
 	// +optional
 	DisabledGatherers []string `json:"disabledGatherers"`
+	// storageSpec is an optional field that allows user to define persistent storage for on-demand gathering
+	// jobs to store the Insights data archive.
+	// If omitted, the gathering job will use ephemeral storage.
+	// +optional
+	StorageSpec *StorageSpec `json:"storageSpec,omitempty"`
+}
+
+// storageSpec provides persistent storage configuration options for on-demand gathering jobs.
+type StorageSpec struct {
+	// type is a required field that specifies the type of storage that will be used to store the Insights data archive.
+	// Valid value is "PersistentVolumeClaim".
```

Comment:
Apologies for the churn here, but if I understood https://github.com/openshift/api/pull/2200#discussion_r1963816147 correctly, we should still allow the explicit setting of `Ephemeral` here. 

In this case it seems like I had a misunderstanding of how the "only one way to say something" convention applies here.